lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1329,6 +1329,41 @@ def test_generate_cert_config(self):
         resp = self._get_progress_page()
         self.assertNotContains(resp, 'Request Certificate')
 
+    @patch('lms.djangoapps.certificates.api.get_active_web_certificate', PropertyMock(return_value=True))
+    @patch.dict('django.conf.settings.FEATURES', {'CERTIFICATES_HTML_VIEW': True})
+    def test_view_certificate_for_unverified_student(self):
+        """
+        If user has already generated a certificate, it should be visible in case of user being
+        unverified too.
+        """
+        certificate = GeneratedCertificateFactory.create(
+            user=self.user,
+            course_id=self.course.id,
+            status=CertificateStatuses.downloadable,
+            download_url="http://www.example.com/certificate.pdf",
+            mode='verified'
+        )
+
+        # Enable the feature, but do not enable it for this course
+        CertificateGenerationConfiguration(enabled=True).save()
+
+        # Enable certificate generation for this course
+        certs_api.set_cert_generation_enabled(self.course.id, True)
+
+        CourseEnrollment.enroll(self.user, self.course.id, mode="verified")
+        with patch('lms.djangoapps.grades.course_grade_factory.CourseGradeFactory.read') as mock_create:
```

Comment:
Before this, you should verify that learner identity verification is expired to make sure this test is hitting our required scenario.

Reply:
If ran like this, it evaluates `_missing_required_verification` as True. Should I also do it explicitly?

Reply:
I'm just saying the user's identity verification should be expired in this test case, so that in future if someone changes the code back to previous state or something else which does not return the certificate for expired identity, this test should fail.